docs/blog/blog.md

```diff
@@ -0,0 +1,221 @@
+
+![Results](./results.png "Results")
+
+Figure 3: Weighted average f1 scores of encoding schemes
+
+## Application
+
+Next, we are going to see how we can apply gamma poisson factorization in an unsupervised setting.
+
+![Dataset](./dataset.png "Data set")
+
+Figure 4: Data set of system hardware configurations
+
+- **Data:** We use a data set of hardware configuration of user systems. Figure 4 shows the system configurations.
+
+- **Problem:** Our aim is to find configurations that are outliers and rules that determine that these configurations are outliers.
+
+- **Encoding:** We use gamma poisson factorization for the variable _modelname_ since it has a high cardinality. For the rest, we use one-hot encoding since they have a low cardinality.
+
+- **Outlier detection:** To find the outliers, we use the isolation forest algorithm that finds points that are easiest to isolate.
+
+- **Interpretation:** Finally, we train a surrogate decision tree based on the labels given by the isolation forest to find rule paths that lead to the outlier classification.
+
+#### **Interpretation**
+
+![Rulepath](./rulepath.png "Rule path")
+
+Figure 5: Example rule path for detecting a system as an anomaly
+
+##### The tree can be interpreted in the following way:
+
+- The nodes represent feature splitting points.
+- The first line represents a condition used to split that node.
+- The second line specifies the percentage of samples that fall in that node.
+- The third line specifies that, for those samples that fall in that node, what percent belong to class Anomaly (orange color) and what % belongs to class Not Anomaly (Blue color)
+
+##### Rule paths
+
+- One-hot encoded variable: if a node says that vendor_seabios &lt;= 0.5, that means that for all the systems where the vendor is not seabios. Since one-hot encoding means 0 or 1, &lt; 0.5 means 0.
+- Gammas Poisson factorized variables: if a node says modelname_broadwell,skylake,haswell &lt;= R (real number), then that means that systems that have strings broadwell, skylake or haswell in their modelname are important factors for splitting the tree further. In order to decide whether their presence or absence causes these anomalies, further validation has to be done.
+- Example rule path found in Figure 4: if (modelname has platinum, pentium, 8175m) and (vendor is seabios) then it could be an anomaly.
```

Comment:
```suggestion
- Example rule path found in Figure 4: if (modelname has _platinum_, _pentium_, _8175m_) and (vendor is _seabios_) then it could be an anomaly.
```